### PR TITLE
[Enhancement] Maintain Imports, and Add Docstring

### PR DESCRIPTION
**Pull request by kevinmatthes**
These changes introduce

* a maintenance of the module import, and
* a new docstring for the syntactic sugar function `matches_filter`.

Since the defined crate has two dependencies defined in its `Cargo.toml`, it might help to import the required modules explicitly.  This not only benefits the readability of the file but also justifies them being listed as dependencies.

Furthermore, since there already is an import line for `std` modules, I added the other components which are requested but not imported at the file's beginning, yet.

The changes are linted and formatted.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -178,7 +181,7 @@ fn main() -> Result<(), Box<dyn Error>> {
     Ok(())
 }
 
-/// A small in-app documentation.
+/// A brief in-app documentation.
```
[the remark this answers is not in the document]
Comment (wert007) [reply src/main.rs:184]: Sure, if you want to 😄

**Comment by wert007** [comment]
First on a more general note. You create quite a lot of commits. E.g. your first 4 commits about the use statements could easily be combined into one. You know, that you can amend to your last commit with `git commit --amend`? Or you could first do all the work and decide later with `git add -p` which parts you actually want to add to a commit.

**Comment by wert007** [comment]
Secondly regarding all the imports at the top. For one in the case of `Duration` there is also a `std::time::Duration`, so by writing `chrono::Duration` it is made more clear which one is used. Similarly for `Options`, since it has such a unspecific name `getopts::Options` is used to clarify what it is and how it should be used. The same holds true for `ParseError` since this is also a very generic name, it probably makes more sense to use `chrono::ParseError`. At that point I would probably also keep fully qualifying `chrono::DateTime`, to make it more obvious, that it correlates to the other `chrono::*` types.

For imports of std modules, I would say, that they make sense, if they are used multiple times. This is not the case for `std::env`.

**Comment by kevinmatthes** [comment]
> First on a more general note. You create quite a lot of commits. E.g. your first 4 commits about the use statements could easily be combined into one. You know, that you can amend to your last commit with `git commit --amend`? Or you could first do all the work and decide later with `git add -p` which parts you actually want to add to a commit.

I commit the changes directly on the website, if possible.  That way, I do not need to upload them and they are ready to merge faster.

I decided to make the commits as atomic as possible to increase the descriptiveness of their messages.  The more atomic they are, the better their commit message can describe what happened.

And yes, I do the work before I commit.  But I test locally and copy all meaningful, optimised and working changes to the website afterwards.

File: src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevinmatthes) [reply src/main.rs:184]: :wrench: :heavy_check_mark:

File: src/main.rs [hunk 2]
```diff
@@ -1,8 +1,11 @@
+use chrono::{DateTime, Duration, FixedOffset, ParseError};
 use getopts::Options;
-use std::{collections::HashMap, error::Error, io::Write, num::ParseIntError, ops::AddAssign};
+use std::{
+    collections::HashMap, env, error::Error, fs, io::Write, num::ParseIntError, ops::AddAssign,
```
Comment (kevinmatthes) [inline src/main.rs:4]: How about changing the `env` import directly to `std::env::args()`? :wrench:

File: src/main.rs [hunk 3]
```diff
@@ -1,8 +1,11 @@
+use chrono::{DateTime, Duration, FixedOffset, ParseError};
```
Comment (kevinmatthes) [inline src/main.rs:1]: So, you basically suggest to completely revert this change? :wrench:

**Comment by wert007** [comment]
> I commit the changes directly on the website, if possible. That way, I do not need to upload them and they are ready to merge faster.
> 
> I decided to make the commits as atomic as possible to increase the descriptiveness of their messages. The more atomic they are, the better their commit message can describe what happened.
> 
> And yes, I do the work before I commit. But I test locally and copy all meaningful, optimised and working changes to the website afterwards.

Okay. But why? Your changes have to be uploaded anyway, I would even argue, that a browser tends to use more data in the process. And time really is not that important for commits, since it is not a race. 😅 

I think you overshoot on the atomic aspect. Like you split a single change over multiple commits. Like adding the `--help` output to the `README` that would be one single commit, but you are splitting it into multiple, since you already commited the change of adding the description of a single command/flag instead. Like adding each flag separately would not count as atomic in my opinion. It would be just too small of a change.

Also if you find missing `,` e.g. you can edit past commits on the command line and just add it there. This is not possible with your workflow.

Sure, that is great, but using `git commit` and `git push` would make so much more sense here..

File: src/main.rs (hunk 2 above)
Comment (wert007) [reply src/main.rs:4]: Hmm, since there is only a single place calling that function, I don't think it is worth it.

Generally I'm not sure how I feel importing functions directly from other modules/libraries. I call them always with their partial namespace at least. 😅

File: src/main.rs (hunk 3 above)
Comment (wert007) [reply src/main.rs:1]: Yes 😄

**Comment by kevinmatthes** [comment]
Okay, I will take your comment into account for future changes.  And again, when merging a Pull Request, you are also allowed to squash commits, and the described problems are gone since there is only one commit with all introduced changes left.

File: src/main.rs (hunk 3 above)
Comment (kevinmatthes) [reply src/main.rs]: :wrench: :heavy_check_mark:

**Comment by wert007** [comment]
The commits are still listed in the commit history, visible in e.g. `git log` 😅

**Comment by kevinmatthes** [comment]
In my opinion, it is reasonable to create single commits for every resolved discussion originating from a review.  For instance, the latest commit just reverts the `chrono` import which I was already able to do because there was nothing left to discuss.  In contrast, the imports of `std` modules I added are still topic of an ongoing discussion.

In general, I consider it more important to commit the changes which were already discussed instead of waiting for the discussion to finish.  Best example:  the documentation of the `--help` option which -- in contrast to the explanatory texts of the other options -- already applied the Imperative / Present.  Since its introduction in #6 was never subject to a discussion, I assumed that there was nothing to discuss, so hence, I added its documentation to the README.  For the other options, there was an ongoing discussion about their mode and tense in #10, so in order to avoid unnecessary commits, I did not add them to the README.

Their mode and tense is now resolved, just the French Spacing is still open.

**Comment by kevinmatthes** [comment]
> The commits are still listed in the commit history, visible in e.g. `git log` sweat_smile

Yes and no:  they are visible in my fork but not in your repository if you squash them.  I would have to delete my fork and to refork your project but that's it.  No obsolete commits in your original repository.

**Comment by wert007** [comment]
Pretty sure, that I did squash your first pull request, but your commit messages are still visible in the main branch 😅

**Comment by wert007** [comment]
> In my opinion, it is reasonable to create single commits for every resolved discussion originating from a review. For instance, the latest commit just reverts the `chrono` import which I was already able to do because there was nothing left to discuss. In contrast, the imports of `std` modules I added are still topic of an ongoing discussion.

Hmm, yes, but I would only count fully resolved discussions and not partially resolved.

> In general, I consider it more important to commit the changes which were already discussed instead of waiting for the discussion to finish. Best example: the documentation of the `--help` option which -- in contrast to the explanatory texts of the other options -- already applied the Imperative / Present. Since its introduction in #6 was never subject to a discussion, I assumed that there was nothing to discuss, so hence, I added its documentation to the README. For the other options, there was an ongoing discussion about their mode and tense in #10, so in order to avoid unnecessary commits, I did not add them to the README.

Hmm, I just think it would be more sensible to have on commit for changing the phrasing of the `--help` output and another for adding it to the README. 😅

File: src/main.rs (hunk 2 above)
Comment (kevinmatthes) [reply src/main.rs:3]: Okay, `env` should be rather removed from the import enumeration.  :wrench:

File: src/main.rs [hunk 4]
```diff
@@ -1,8 +1,8 @@
 use getopts::Options;
-use std::{collections::HashMap, error::Error, io::Write, num::ParseIntError, ops::AddAssign};
+use std::{collections::HashMap, error::Error, fs, io::Write, num::ParseIntError, ops::AddAssign};
 
 fn main() -> Result<(), Box<dyn Error>> {
-    let mut opts = getopts::Options::new();
+    let mut opts = Options::new();
```
Comment (wert007) [inline src/main.rs:5]: I already wrote a comment about it, but maybe it got lost. I think `Options` is a too generic name for the struct. So I prefer `getopts::Options`. Otherwise it might be less clear, what is happening here.

Comment (kevinmatthes) [reply src/main.rs]: :wrench: :heavy_check_mark:

File: src/main.rs [hunk 5]
```diff
@@ -1,81 +1,80 @@
...
     let mut opts = getopts::Options::new();
     let opts = opts
-        .optflag("q", "quiet", "Hides the output of commit messages.")
+        .optflag("q", "quiet", "Hide the output of commit messages.")
         .optflag("h", "help", "Show this help and exit.")
         .optmulti(
             "a",
             "author-contains",
-            "Filters after certain author names. ORs if multiple specified.",
+            "Filter for certain author names. ORs if multiple specified.",
```
Comment (kevinmatthes) [inline src/main.rs:11]: "Filter after" seemed to be "not the yellow from the egg"; it means here the temporal order of something.  Since it was obviously intended to describe the extraction of information by a certain filter criterion, I changed it to "filter for".

File: src/main.rs [hunk 6]
```diff
@@ -1,81 +1,80 @@
...
             "file-extension",
-            "Filters loc after certain file extension (e.g. `--file-extension cpp`). ORs if multiple specified.",
+            "Filter loc for certain file extension (e.g. `--file-extension cpp`). ORs if multiple specified.",
             "EXTENSION",
         )
         .optopt(
             "d",
             "duration",
-            "The time, which may pass between two commits, that still counts as working.",
+            "The time which may pass between two commits that still counts as working.",
```
Comment (kevinmatthes) [inline src/main.rs:71]: The commas were not required there.

**Review by wert007 — COMMENTED** [review]
Great, I like it.

File: src/main.rs [hunk 7]
```diff
@@ -1,81 +1,80 @@
...
             "HOURS",
         )
         .optopt(
             "o",
             "output",
-            "An output file for the commits per day in csv format.",
+            "An output file for the commits per day in CSV format.",
```
Comment (kevinmatthes) [inline src/main.rs:77]: File formats are usually named in capital letters.

**Review by kevinmatthes — COMMENTED** [review]
I fixed the language issues described in #10.  Any feedback or anything else?

File: src/main.rs (hunk 5 above)
Comment (wert007) [reply src/main.rs:11]: Yeah, prepositions are my Achilles heal 😅

**Comment by wert007** [comment]
> I fixed the language issues described in #10. Any feedback or anything else?

No, only the `--help` output in the README is left I think.

File: README.md [hunk 8]
```diff
@@ -26,3 +26,10 @@ commit-analyzer outputfile.txt --output "commits-and-loc-by-date.csv"
 
 This project's license is **MIT**.  The license can be found `LICENSE` in the
 main directory of this repository.
+
+## Configured Options
+
+| Option    | Brief | Meaning                                                  |
+|:---------:|:-----:|:---------------------------------------------------------|
+| `--help`  | `-h`  | Show this help and exit.                                 |
+
```
Comment (kevinmatthes) [inline README.md:35]: Revert or enrich with further options? :wrench:

Comment (wert007) [reply README.md:35]: Definitely enriching. I think it was premature to commit the `--help` flag, but reverting would not solve that, and I'm still in favor of adding the output to the README. 😄

Comment (kevinmatthes) [reply README.md]: :wrench: :heavy_check_mark:

File: src/main.rs (hunk 5 above)
Comment (kevinmatthes) [reply src/main.rs]: Just for completeness:  "heel" (part of the leg), not "heal" (to cure).
